### PR TITLE
7.1 cg automation updates

### PR DESCRIPTION
**Pull request by sumabai**
# Description

This PR includes 7.1 CG Functional test script enhancements, Functional suite development, Systemic test automation including stress and scale tests, Systemic test suite.

Jira tasks:
Functional tests : https://issues.redhat.com/browse/RHCEPHQE-13926
Systemic tests : https://issues.redhat.com/browse/RHCEPHQE-13851

**Files:**
Conf file for Systemic suite : conf/reef/cephfs/tier-1_cephfs_cg_quiesce.yaml
Functional suite : suites/reef/cephfs/tier-1_cephfs_cg_quiesce.yaml
Systemic suite : suites/reef/cephfs/tier-1_cephfs_cg_quiesce_systemic.yaml

TestLib:  tests/cephfs/snapshot_clone/cephfs_cg_io.py

Test Scripts:
tests/cephfs/snapshot_clone/cg_snap_test.py
tests/cephfs/snapshot_clone/cg_snap_system_test.py

Logs:
Stress test : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-P4WY12/cg_snap_system_test_0.log
Scale test : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MRROTV/cg_snap_system_test_0.log
Functional tests:
Workflow1 - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-K7ZJUW/cg_snap_test_0.log
Workflow2 - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-LOAL67/cg_snap_test_0.log
<There is cleanup failure in workflow2 because subvolgroup_cg has other test volumes used in other tests. On a fresh subvolgroup, this issue will not be seen>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/snapshot_clone/cg_snap_system_test.py [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+                assert False, f"Test {test_name} failed"
+            else:
+                log.info(f"Test {test_name} passed \n")
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean Up in progess")
```
Comment (neha-gangadhar) [inline tests/cephfs/snapshot_clone/cg_snap_system_test.py:153]: Can you please check finally block? I do not see any nfs cluster nad export created in the test.

Comment (sumabai) [reply tests/cephfs/snapshot_clone/cg_snap_system_test.py:158]: done

**Review by AmarnatReddy — COMMENTED** [review]
LGTM

**Review by hkadam134 — APPROVED** [review]
LGTM

**Review by AmarnatReddy — APPROVED** [review]

**Pull request merged**